File: tests/testdata/codegen/string-enum-union-list-pp/nodejs/string-enum-union-list.ts
import * as khulnasoft from "@khulnasoft/khulnasoft";
import * as azure_native from "@khulnasoft/azure-native";

const namespaceAuthorizationRule = new azure_native.servicebus.NamespaceAuthorizationRule("namespaceAuthorizationRule", {
    authorizationRuleName: "sdk-AuthRules-1788",
    namespaceName: "sdk-Namespace-6914",
    resourceGroupName: "ArunMonocle",
    rights: [
        azure_native.servicebus.AccessRights.Listen,
        azure_native.servicebus.AccessRights.Send,
    ],
});

File: tests/testdata/codegen/string-enum-union-list-pp/go/string-enum-union-list.go
package main

import (
	"github.com/khulnasoft/khulnasoft-azure-native/sdk/go/azure/servicebus"
	"github.com/khulnasoft/khulnasoft/sdk/v3/go/khulnasoft"
)

func main() {
	khulnasoft.Run(func(ctx *khulnasoft.Context) error {
		_, err := servicebus.NewNamespaceAuthorizationRule(ctx, "namespaceAuthorizationRule", &servicebus.NamespaceAuthorizationRuleArgs{
			AuthorizationRuleName: khulnasoft.String("sdk-AuthRules-1788"),
			NamespaceName:         khulnasoft.String("sdk-Namespace-6914"),
			ResourceGroupName:     khulnasoft.String("ArunMonocle"),
			Rights: khulnasoft.StringArray{
				khulnasoft.String(servicebus.AccessRightsListen),
				khulnasoft.String(servicebus.AccessRightsSend),
			},
		})
		if err != nil {
			return err
		}
		return nil
	})
}
